sky/skylet/providers/gcp/node.py

```diff
@@ -621,6 +630,54 @@ def delete_instance(self, node_id: str, wait_for_operation: bool = True) -> dict
 
         return result
 
+    def resize_disk(
+        self, base_config: dict, instance_name: str, wait_for_operation: bool = True
+    ) -> dict:
+        """Resize a Google Cloud disk based on the provided configuration."""
+
+        # Extract the specified disk size from the configuration
+        new_size_gb = base_config["disks"][0]["initializeParams"]["diskSizeGb"]
+
+        # Fetch the instance details to get the disk name and current disk size
+        response = (
+            self.resource.instances()
+            .get(
+                project=self.project_id,
+                zone=self.availability_zone,
+                instance=instance_name,
+            )
+            .execute()
+        )
+        disk_name = response["disks"][0]["source"].split("/")[-1]
+
+        try:
+            # Execute the resize request and return the response
+            operation = (
+                self.resource.disks()
+                .resize(
+                    project=self.project_id,
+                    zone=self.availability_zone,
+                    disk=disk_name,
+                    body={
+                        "sizeGb": str(new_size_gb),
+                    },
+                )
+                .execute()
+            )
+        except HttpError as e:
+            # Catch HttpError when provided with invalid value for new disk size.
+            # Raises error when the new disk size is smaller than the exisiting
+            # disk size
+            logger.warning(f"googleapiclient.errors.HttpError: {e.reason}")
+            return {}
```

Comment:
Why do we need this check? IIUC we already checked this in `resources._try_validate_image_id`?

Reply:
Thanks for the suggestion. I've removed the check here and updated `resources._try_validate_image_id` to ensure that the given disk size is strictly larger than the machine image size. 

In fact, the previous implementation would generate the following error during runtime without the check if the specified disk size is the same as the machine image size:
`returned "Invalid value for field 'sizeGb': '150'. New disk size '150' GiB must be larger than existing size '150' GiB.". Details: "[{'message': "Invalid value for field 'sizeGb': '150'. New disk size '150' GiB must be larger than existing size '150' GiB.", 'domain': 'global', 'reason': 'invalid'}]"`

Reply:
Good catch! Thx!